doc/feature/synapses.md

```diff
@@ -0,0 +1,215 @@
+    float preDistance() const;
+    Vector3f preSurfacePosition() const;
+    Vector3f preCenterPosition() const;
+
+    uint32_t postGID() const;
+    uint32_t postSectionID() const;
+    uint32_t postSegmentID() const;
+    float postDistance() const;
+    Vector3f postSurfacePosition() const;
+    Vector3f postCenterPosition() const;
+
+    SynapseGID gid() const;
+    float conductance() const;
+    float utilization() const;
+    float depression() const;
+    float facilitation() const;
+    float decay() const;
+    int efficacy() const;
+};
+
+// read-only, moveable, non-copyable
+class Synapses
+{
+    // conversion ctor; read all synapses from selected GIDs into memory
+    Synapses( const SynapsesStream& );
+
+    size_t size() const;
+    bool empty() const;
+
+    bool operator==( const Synapses& rhs ) const;
+    bool operator!=( const Synapses& rhs ) const;
+
+    const_iterator begin() const;
+    const_iterator end() const;
+
+    const Synapse& operator[]( size_t ) const;
+
+    const size_t* index() const;
+    const uint32_t* preGID() const;
+    const uint32_t* preSectionID() const;
+    const uint32_t* preSegmentID() const;
+    const float* preDistance() const;
+    const float* preSurfacePositionX() const;
+    const float* preSurfacePositionY() const;
+    const float* preSurfacePositionZ() const;
+    const float* preCenterPositionX() const;
+    const float* preCenterPositionY() const;
+    const float* preCenterPositionZ() const;
+
+    const uint32_t* postGID() const;
+    const uint32_t* postSectionID() const;
+    const uint32_t* postSegmentID() const;
+    const float* postDistance() const;
+    const float* postSurfacePositionX() const;
+    const float* postSurfacePositionY() const;
+    const float* postSurfacePositionZ() const;
+    const float* postCenterPositionX() const;
+    const float* postCenterPositionY() const;
+    const float* postCenterPositionZ() const;
+
+    const size_t* indices() const;
+    const float* conductance() const;
+    const float* utilization() const;
+    const float* depression() const;
+    const float* facilitation() const;
+    const float* decay() const;
+    const int* efficacy() const;
+};
+
+// sequentially & forwarding iteration through synapses for any number of GIDs
+class SynapsesStream
+{
+    bool eos() const;
+
+    // - async load of synapses for count cells
+    // - not order guarantee
+    // - result Synapses can be empty
+    // - throws if current + count leads to eos
+    std::future< Synapses > read( size_t count = 1 ) const;
+};
+
+```
+
+## File format
+
+Uses existing nrn* files
+
+## Examples
+
+### Stream synapse positions for pathway density voxelization
+
+```cpp
+const brain::Circuit circuit;
+const brain::SynapsesStream& stream =
+   circuit.getProjectedSynapses( circuit.getGIDs( "mtypetarget_1" ),
+                                 circuit.getGIDs( "mtypetarget_2" ));
+std::future< brain::Synapses > future = stream.read();
```

Comment:
I still would like to know how can I know which is the largest value I can pass to read in the stream returned by getProjectedSynapses.